### ISSUE TITLE
After loss.backwards() the model frozes

### ISSUE THREAD

**Issue by kratosaba27** [pytorch/pytorch#21143]
## 🐛 Bug

I'm trying to run a simple model, from an example from the pytorch [website](https://pytorch.org/tutorials/beginner/pytorch_with_examples.html). However I encounter a really strange bug, when I run it the program stops. I added some line prints to see where exactly stops and is in the loss.backwards(). I have the same issue with a model that I created.  

## To Reproduce

1. Not really sure how to reproduce, I've tried to reinstall pytorch and still persist. 

```
import torch

# N is batch size; D_in is input dimension;
# H is hidden dimension; D_out is output dimension.
N, D_in, H, D_out = 64, 1000, 100, 10

# Create random Tensors to hold inputs and outputs
x = torch.randn(N, D_in)
y = torch.randn(N, D_out)

# Use the nn package to define our model as a sequence of layers. nn.Sequential
# is a Module which contains other Modules, and applies them in sequence to
# produce its output. Each Linear Module computes output from input using a
# linear function, and holds internal Tensors for its weight and bias.
model = torch.nn.Sequential(
    torch.nn.Linear(D_in, H),
    torch.nn.ReLU(),
    torch.nn.Linear(H, D_out),
)

# The nn package also contains definitions of popular loss functions; in this
# case we will use Mean Squared Error (MSE) as our loss function.
loss_fn = torch.nn.MSELoss(reduction='sum')

learning_rate = 1e-4
for t in range(500):
    # Forward pass: compute predicted y by passing x to the model. Module objects
    # override the __call__ operator so you can call them like functions. When
    # doing so you pass a Tensor of input data to the Module and it produces
    # a Tensor of output data.
    y_pred = model(x)

    # Compute and print loss. We pass Tensors containing the predicted and true
    # values of y, and the loss function returns a Tensor containing the
    # loss.
    loss = loss_fn(y_pred, y)
    print(t, loss.item())
    print("is fine")
    # Zero the gradients before running the backward pass.
    model.zero_grad()
    print("is fine")

    # Backward pass: compute gradient of the loss with respect to all the learnable
    # parameters of the model. Internally, the parameters of each Module are stored
    # in Tensors with requires_grad=True, so this call will compute gradients for
    # all learnable parameters in the model.
    loss.backward()
    print("is fine")

    # Update the weights using gradient descent. Each parameter is a Tensor, so
    # we can access its gradients like we did before.
    with torch.no_grad():
        for param in model.parameters():
            param -= learning_rate * param.grad
```
## Expected behavior
Just run the network as expected. 

## Environment

    PyTorch Version > 1.1.0
    Ubuntu: 18
    How I installed PyTorch: conda
    Python version: 3
    CUDA/cuDNN version: V9.1.85
    GPU models and configuration: Nvidia GTX 1050 Ti

You can see what I mean in this image. 
![Screenshot from 2019-05-30 14-50-58](https://user-images.githubusercontent.com/44777055/58634026-82cde300-82ea-11e9-98e6-ad52541e736b.png)

**Comment by Genius1237** [comment]
I've tried to reproduce this with python from ubuntu repos and python from conda. I am not able to. Can you provide more details (finer details) about which versions of `python`, `torch` and where you've installed it from

**Comment by kratosaba27** [comment]
I installed it from the command line in Ubuntu, you can check the version above. using anaconda

**Comment by ezyang** [comment]
Can you run your program under gdb? When it hangs, C-c it and send us the backtraces of as many threads as you can.

**Comment by ezyang** [comment]
I ran the script a few times on a source build of e161360b62e39e3a22175a2b08b0a4a20b3aee77 on Ubuntu 18.04 but could not reproduce. Another thing you could try is installing a different distribution (e.g., pip) or compiling from source.

**Comment by Genius1237** [comment]
To add to this, I have tested this with version `1.1.0` cpu from pip with `python 3.6` on 18.04 and version `0.4.1post2` cuda from conda with `python 3.6` on 18.04 and cannot reproduce it.

**Comment by kratosaba27** [comment]
Ok tried to install it with pip, can you guys tell me how to do it from source?

**Comment by Genius1237** [comment]
Does the problem persist with the installation from `pip`?

Have you ever tried installing any other package from source before? The instructions are there here https://github.com/pytorch/pytorch#from-source. It'll take a while to compile, it should eventually be done.

**Comment by kratosaba27** [comment]
Yeah the problem persist with pip. 

I just remember that I have an extra library https://github.com/fwilliams/fml. Maybe this might help to reproduce the problem. 

If you don't seem to reproduce it I'll try to install from souce.

**Comment by Genius1237** [comment]
In the code sample you provided, the `fml` library isn't imported. I installed the library from `conda-forge`, I imported it and ran your code sample, and it runs properly.

If you feel that some library is the problem, try uninstalling it and running your sample code.

Have you also tried creating a new `conda` environment, installing only pytorch's necessary packages and running this sample?

**Comment by kratosaba27** [comment]
Ok, I'll try to create a new enviorment and installing everything there.

**Comment by kratosaba27** [comment]
Ok I tried to do that and it didn't help. Just to be sure that I'm uninstalling pytorrch correctly. I'm using this command: 

conda uninstall pytorch torchvision cuda80 -c soumith.

**Comment by Genius1237** [comment]
Create a new environment using
`conda create -n yourenvname python=3.6 anaconda`
Then switch to that env via
`conda activate yourenvname`
Then install pytorch
`conda install pytorch torchvision cudatoolkit=9.0 -c pytorch`

**Comment by kratosaba27** [comment]
I did just that and it didn't work again :/. I don't know what I'm doing wrong D:

**Comment by kratosaba27** [comment]
Finally I had to re install ubuntu and re install pytorch, it seems that I had a bigger problem with ubuntus installations. Thank you for your help
